examples/cloudflare-edge/server.ts

```diff
@@ -0,0 +1,229 @@
+import * as build from '@remix-run/dev/server-build';
+import {createCookieSessionStorage, ServerBuild} from '@remix-run/cloudflare';
+import type {GetLoadContextFunction} from '@remix-run/cloudflare-workers';
+import {
+  createRequestHandler as createRemixRequestHandler,
+  Session,
+  SessionStorage,
+} from '@remix-run/server-runtime';
+import type {Options as KvAssetHandlerOptions} from '@cloudflare/kv-asset-handler';
+import {
+  MethodNotAllowedError,
+  NotFoundError,
+  getAssetFromKV,
+} from '@cloudflare/kv-asset-handler';
+import {createStorefrontClient} from '@shopify/hydrogen';
+import {getBuyerIp} from '@shopify/remix-oxygen';
+
+type RequestHandler = (event: FetchEvent) => Promise<Response>;
+
+addEventListener(
+  'fetch',
+  createEventHandler({build, mode: process.env.NODE_ENV}),
+);
+
+function createEventHandler({
+  build,
+  getLoadContext,
+  mode,
+}: {
+  build: ServerBuild;
+  getLoadContext?: GetLoadContextFunction;
+  mode?: string;
+}) {
+  let handleEvent = async (event: FetchEvent) => {
+    /**
+     * Open a cache instance in the worker and a custom session instance.
+     */
+    if (!SESSION_SECRET) {
+      throw new Error('SESSION_SECRET environment variable is not set');
+    }
+
+    const waitUntil = (p: Promise<any>) => event.waitUntil(p);
+    const env = {
+      PUBLIC_STOREFRONT_API_TOKEN,
+      PRIVATE_STOREFRONT_API_TOKEN:
+        typeof PRIVATE_STOREFRONT_API_TOKEN !== 'undefined'
+          ? PRIVATE_STOREFRONT_API_TOKEN
+          : undefined,
```

Comment:
I was having issues with env variable bindings not existing. There is probably a better way of doing this.